glide-core/src/client/value_conversion.rs

```diff
@@ -379,6 +380,101 @@ pub(crate) fn convert_to_expected_type(
             )
                 .into()),
         },
+        // Used by GEOSEARCH.
+        // When all options are specified (withcoord, withdist, withhash) , the response looks like this: [[name (str), [dist (str), hash (int), [lon (str), lat (str)]]]] for RESP2.
+        // RESP3 return type is: [[name (str), [dist (str), hash (int), [lon (float), lat (float)]]]].
+        // We also want to convert dist into float.
+        /* from this:
+        > GEOSEARCH Sicily FROMLONLAT 15 37 BYBOX 400 400 km ASC WITHCOORD WITHDIST WITHHASH
+        1) 1) "Catania"
```

Comment:
Because I wanted it to be a list of object for Java

We can't split this function into one with just the names and one with options because user can pass False to all options
And making the options an enum is not what we want to do, it's not a nice api

Reply:
We should include an example without options. 
Since the result is a different data structure.

Reply:
On the Java-side, we could have the options require that at least one of WITHCOORD, WITHDIST, or WITHHASH is selected.  Otherwise, they can have a simple Array return.

Reply:
is it possible on all languages? I think we should not risk it